internal/adapter/adapter.go

```diff
@@ -68,24 +68,32 @@ func (b *Bot) GetFirstName() string {
 
 func (b *Bot) adaptUpdates(updates <-chan tgbotapi.Update, messages chan<- *Message) {
 	var err error
+	var msg *tgbotapi.Message
 	for update := range updates {
+		if update.Message != nil {
+			msg = update.Message
+		} else {
+			msg = update.ChannelPost
+		}
+
 		message := &Message{
 			Replies: make(chan *Message),
-			From:    update.Message.From.UserName,
-			ChatID:  update.Message.Chat.ID,
+			From:    msg.From.UserName,
```

Comment:
No, according to offcial API documentation https://core.telegram.org/bots/api#message:
`from ... Optional. Sender, can be empty for messages sent to channels`